src/main/java/mil/dds/anet/database/ReportSensitiveInformationDao.java

```diff
@@ -148,7 +148,13 @@ private boolean isAuthorized(Person user, Report report) {
 		// Check the organization for which the user is authorized
 		final Organization userOrg = userPosition.loadOrganization();
 		final Organization advisorOrg = report.loadAdvisorOrg();
-		if (userOrg != null && advisorOrg != null && DaoUtils.getId(userOrg) == DaoUtils.getId(advisorOrg)) {
+		if (userOrg == null || advisorOrg == null) {
+			// No organization
+			return false;
+		}
+		final Integer userOrgId = DaoUtils.getId(userOrg);
+		final Integer advisorOrgId = DaoUtils.getId(advisorOrg);
+		if (userOrgId != null && userOrgId.equals(advisorOrgId)) {
```

Comment:
Minor point, maybe use the static Integer.compare(x, y)

Reply:
That would mean I'd have to do an additional check that `advisorOrgId != null` since `Integer.compare` unboxes both arguments.